graph/warshall_transitive_closure.cpp

```diff
@@ -0,0 +1,179 @@
+/**
+* @file
+* @brief Implementation of Warshall's Algorithm in C++
+* [Warshall's Transitive Closure Algorithm](https://www.javatpoint.com/warshalls-algorithm)
+* 
+* @details
+* Warshall's Algorithm is used to determine
+* the transitive closure of a directed graph 
+* or all paths in a directed graph by using the adjacency matrix. 
+* For this, it generates a sequence of n matrices.
+* Where, n is used to describe the number of vertices. 
+*
+* i.e.,
+* R(0), ..., R(k-1), R(k), ... , R(n) 
+*
+* In this program, maximum value of nodes 
+* that can be taken under consideration is 10
+* (which is changeable)
+*
+* Reason: In Warshall's algo, Θ(n^3) is the worst-case cost.
+* Even Brute Force method seems a viable option when compared
+* to this. Therefore static 2D array is used with fixed
+* input size of 10 inputs in this program in order to demonstrate
+* the implementation and maintain responsiveness. 
+*
+* The time complexity of this algorithm is O(n^3)
+*
+* In order to generate R(k) from R(k-1),
+* the following rules will be implemented:
+*
+* Rule 1: In row i and column j, 
+* if the element is 1 in R(k-1), 
+* then in R(k), it will also remain 1.
+*
+* Rule 2: In row i and column j,
+* if the element is 0 in R(k-1), 
+* then the element in R(k) will be changed to 1 
+* iff the element in its column j and row k,
+* and the element in row i and column k are both 1's in R(k-1). 
+*
+* The resulting matrix after following the steps n times and obtaining
+* n number of matrices, the nth matrix is the transitive closure for
+* the given adjacency matrix.
+*
+* @author [Abhilipsa Sahoo](https://github.com/abhilipsasahoo03)
+*/
+
+
+#include<iostream> //For IO Operations
+#include<cassert> //For using assert()
+
+
+/** Automating tests to check if the algorithm runs perfectlyperfectly
+* for two test cases
+* @return None
+*/
+static void test()
+{
+  //Test 1
+  int Arr1[10][10] = {{1, 0, 0, 1}, {0, 1, 1, 0}, {0, 0, 0, 1}, {0, 0, 0, 0}};
```

Comment:
Due to the use of static/C-style arrays, the CI is specifying we should use `std::array` or `std::vector` instead.
This should be still implementable and should work fine with this algorithm. If you need any help, let us know. 🙂